pkg/services/accesscontrol/database/database.go

```diff
@@ -62,71 +89,84 @@ func (s *AccessControlStore) SearchUsersPermissions(ctx context.Context, orgID i
 	}
 	dbPerms := make([]UserRBACPermission, 0)
 
+	var userID int64
+	if options.NamespacedID != "" {
+		var err error
+		userID, err = options.ComputeUserID()
+		if err != nil {
+			return nil, err
+		}
+	}
+
 	if err := s.sql.WithDbSession(ctx, func(sess *db.Session) error {
 		roleNameFilterJoin := ""
 		if len(options.RolePrefixes) > 0 {
-			roleNameFilterJoin = "INNER JOIN role AS r on up.role_id = r.id"
+			roleNameFilterJoin = "INNER JOIN role AS r ON up.role_id = r.id"
+		}
+
+		params := []any{}
+
+		direct := userAssignsSQL
+		if options.NamespacedID != "" {
+			direct += " WHERE ur.user_id = ?"
+			params = append(params, userID)
+		}
+
+		team := teamAssignsSQL
+		if options.NamespacedID != "" {
+			team += " WHERE tm.user_id = ?"
+			params = append(params, userID)
+		}
+
+		basic := basicRoleAssignsSQL
+		if options.NamespacedID != "" {
+			basic += " WHERE ou.user_id = ?"
+			params = append(params, userID)
+		}
+
+		grafanaAdmin := fmt.Sprintf(grafanaAdminAssignsSQL, s.sql.Quote("user"))
+		params = append(params, accesscontrol.RoleGrafanaAdmin)
+		if options.NamespacedID != "" {
+			grafanaAdmin += " AND sa.user_id = ?"
+			params = append(params, userID)
 		}
 
 		// Find permissions
 		q := `
 		SELECT
```

Comment:
I like the approach of pulling out the subqueries so that it's easier to reason about how this query is constructed. Not sure if this is just personal preference, but this overall query could also be parameterized and made a constant (using `fmt.Sprintf()`). Take it or leave it though, I don't think it's critical.

Reply:
Thanks :)
I have a slight preference over with this approach, but no strong opinion as well :+1: